src/postgres/mod.rs

```diff
@@ -61,6 +67,52 @@ impl Postgres {
             .insert("POSTGRES_PASSWORD".to_owned(), password.to_owned());
         self
     }
+
+    /// Registers sql to be executed automatically when the container starts.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use testcontainers_modules::postgres::Postgres;
+    /// let postgres_image = Postgres::default()
+    ///                                .with_init_sql("CREATE EXTENSION IF NOT EXISTS hstore;");
+    /// # assert!(postgres_image.is_ok())
+    /// ```
+    ///
+    /// ```rust,ignore
+    /// # use testcontainers_modules::postgres::Postgres;
+    /// let postgres_image = Postgres::default()
+    ///                                .with_init_sql(include_str!("path_to_init.sql"));
+    /// # assert!(postgres_image.is_ok())
+    /// ```
+    pub fn with_init_sql(mut self, init_sql: &str) -> Result<Self, std::io::Error> {
+        let init_path = match self.tempdir.as_ref() {
+            Some(t) => t
+                .path()
+                .join(format!("init_{index}.sql", index = self.init_mounts.len())),
+            None => {
+                return Err(std::io::Error::other(
+                    "Could not create a temporary directory",
+                ))
+            }
+        };
+        let mut tmp_file = File::create(init_path.clone())?;
+        write!(tmp_file, "{}", init_sql)?;
+        // convert the init_path to a file mount
+        let file_name = init_path
+            .file_name()
+            .expect("we created files with filenames, not directories")
+            .to_str()
+            .expect("we created files with filenames, not directories");
+        let container_path = format!("/docker-entrypoint-initdb.d/{file_name}",);
+        self.init_mounts.push(Mount::bind_mount(
+            init_path
+                .to_str()
+                .expect("the path we put is a valid string"),
+            container_path,
+        ));
+        Ok(self)
+    }
```

Comment:
The similar approach can be implemented for some others DB modules (e.g mysq, mariadb)

Also, important part here: it uses blocking API and might be called from async code.

I think we need to have more general interface for these modules, with ability to run async and sync (we can spawn a blocking task at least)

I can suggest to extract a trait for RDBMS (e.g `InitSql` and `InitSqlAsync`). Also this common module may have different strategies: `TempFile`, `Sqlx` and etc. 
For now we can implement only the one on top of `TempFile`, it will create a temp directory/file and return a structure, like `TemporaryScript { tempdir: TempDir, path: PathBuf }`
So `Postgres` module will need only to mount this and keep the structure for it's lifetime


___
Probably it's overwhelming comment and not well described. Feel free to clarify anything and share your opinion

Reply:
Might have misunderstood something ^^

This could work similarly, to the [wait's-stratgy patern](https://github.com/testcontainers/testcontainers-rs/tree/main/testcontainers/src/core/wait)..

```rust
#[derive(Default)]
enum InitSql {
  #[default]
  None,
  TemporaryDirectory(TemporaryDirectoryStrategy), // deliberatley not a TempFile, as this way multiple files are supported
  Sqlx(SqlxStrategy),
}

impl InitSql{
  pub fn mount_temporary_files(TemporaryDirectoryStrategy)
}

// InitSqlAsync is the same but with async in all relevant names
```

Not quite sure what a `InitSqlStrategy` trait would look like.
Any draft that I came up with shifts a lot of the work onto the RDMS..

Running sqlx commands after something is healthy and mounting files is not quite similar.
=> Not sure if marrying them at the hip as above is a good idea.. `TemporaryDirectory` needs a base path to retunrn the mounts while `Sqlx` needs a healthy connectionstring/PGConnection/...